mailpoet/assets/js/src/analytics.js

```diff
@@ -12,3 +12,51 @@ export function getTrackingData() {
   }
   return trackingDataLoading;
 }
+
+export function mapFilterType(filter) {
+  const action = filter.action;
+  const filterType = filter.type;
+
+  if (filterType === 'email' && action === 'machineOpensAbsoluteCount')
+    return '# of machine-opens';
+  if (filterType === 'email' && action === 'opensAbsoluteCount')
+    return '# of opens';
+  if (filterType === 'woocommerce' && action === 'numberOfOrders')
+    return '# of orders';
+  if (filterType === 'email' && action === 'clicked') return 'clicked';
+  if (filterType === 'email' && action === 'clickedAny')
+    return 'clicked any email';
+  if (filterType === 'userRole' && action === 'subscriberScore') return 'score';
+  if (filterType === 'userRole' && action === 'subscribedToList')
+    return 'subscribed to list';
+  if (filterType === 'email' && action === 'opened') return 'opened';
+  if (filterType === 'email' && action === 'machineOpened')
+    return 'machine-opened';
+  if (filterType === 'woocommerceMembership' && action === 'isMemberOf')
+    return 'is active member of';
+  if (
+    filterType === 'woocommerceSubscription' &&
+    action === 'hasActiveSubscription'
+  )
```

Comment:
I would like to believe there's a better way to handle this block of code, but my brain is too tired to think right now.

I do think it makes readability a little challenging and makes updates tricky.

A quick suggestion I may offer:

Try to align and order by the same type of `filterType`
e.g
if filterType === 'email'

Show all filterType === 'email' before showing filterType === 'userRole' or filterType === 'woocommerce'

In this case, we have filterType === 'email', filterType === 'woocommerce' then filterType === 'email' again.

Reply:
Sorry for jumping in, in these cases I really like the pattern used for [creating reducers](https://github.com/kolodny/redux-create-reducer#readme) which is used for [NgRX reducers](https://ngrx.io/guide/store/reducers) as well, which is basically only a hash of `case => reducers functions`, it's only a suggestion and maybe too big of a change for this case!